tests/integration/test_save_results_in_table.py

```diff
@@ -0,0 +1,76 @@
+import pytest
```

Comment:
Please move this to unit tests (`tests/unit` folder). I would generally avoid extensive mocking as it makes refactoring later difficult. Also please don't relay on private methods for testing.

For integration testing please setup a proper dataframe and make sure the data is actually saved in the tables as defined in the config.